src/fluree/store/core.cljc

```diff
@@ -0,0 +1,69 @@
+(ns fluree.store.core
```

Comment:
I think we should move away from `*core` namespace names, and this should be under `fluree.db`. 

There's already a storage namespace. The protocol should probably live there.